operators/derive/to_es.py

```diff
@@ -0,0 +1,68 @@
+import dataflows as DF
+from dataflows.helpers.resource_matcher import ResourceMatcher
+from dataflows_elasticsearch import dump_to_es
+
+from conf import settings
+from srm_tools.logger import logger
+
+
+def set_es_item_types(resources=None):
+    def func(package):
+        matcher = ResourceMatcher(resources, package.pkg)
+        for resource in package.pkg.descriptor['resources']:
+            if matcher.match(resource['name']):
+                for field in resource["schema"]["fields"]:
+                    if field["type"] == "any":
+                        field["type"] = "string"
+                    elif field["type"] == "array":
+                        field["es:itemType"] = "string"
+        yield package.pkg
+
+        res_iter = iter(package)
+        for r in res_iter:
+            if matcher.match(r.res.name):
+                yield r.it
+            else:
+                yield r
+
+    return func
+
+
+def data_api_es_flow():
+    return DF.Flow(
+        DF.load(f'{settings.DATA_DUMP_DIR}/table_data/datapackage.json'),
+        DF.printer(),
+        # the tableschema-elasticsearch driver requires
+        # a primary key but elasticsearch does not.
```

Comment:
primary key is used to create the ids for the documents - so that each document gets a unique id

Reply:
Sure, but the question is why - in the SQL driver, if there is no primary key, we let the backend autogenerate ids, and for a search index that seems an even more appropriate solution.

Reply:
I think the answer is related to the update flow - 
with SQL you do an 'update <fields> where <conditions>' kind of query, basically filtering the data itself. So you don't even need the autogenerated ids - you just find the relevant row based on its contents.
with ES it's simpler to do an update to the document by id, and not by querying the data (although it's also possible, but requires specific indexing of the data).

Reply:
got it